core/src/main/java/edu/wpi/grip/core/operations/opencv/NewKernelOperation.java

```diff
@@ -0,0 +1,89 @@
+package edu.wpi.grip.core.operations.opencv;
+
+
+import edu.wpi.grip.core.Description;
+import edu.wpi.grip.core.OperationDescription;
+import edu.wpi.grip.core.sockets.InputSocket;
+import edu.wpi.grip.core.sockets.OutputSocket;
+import edu.wpi.grip.core.sockets.SocketHint;
+import edu.wpi.grip.core.sockets.SocketHints;
+
+import com.google.inject.Inject;
+
+import org.bytedeco.javacpp.opencv_core.Mat;
+import org.bytedeco.javacpp.opencv_core.Size;
+import org.bytedeco.javacpp.opencv_imgproc;
+import org.python.google.common.collect.ImmutableList;
+
+import java.util.List;
+
+@Description(name = "New Kernel",
+        summary = "Create a kernel of custom size",
+        category = OperationDescription.Category.OPENCV,
+        iconName = "kernel")
+public class NewKernelOperation implements CVOperation {
+
+  private final SocketHint<KernelEnum> typeHint = SocketHints.createEnumSocketHint("kernelType",
+          KernelEnum.MORPH_RECT);
+  private final SocketHint<Number> widthHint = SocketHints.Inputs
+          .createNumberSpinnerSocketHint("width", 1, 1, Integer.MAX_VALUE);
+  private final SocketHint<Number> heightHint = SocketHints.Inputs
+          .createNumberSpinnerSocketHint("height", 1, 1, Integer.MAX_VALUE);
+  private final SocketHint<Mat> outputHint = SocketHints.Outputs.createMatSocketHint("kernel");
+
+
+  private final InputSocket<Number> widthSocket;
+  private final InputSocket<Number> heightSocket;
+  private final InputSocket<KernelEnum> typeSocket;
+
+  private final OutputSocket<Mat> outputSocket;
+
+  @Inject
+  @SuppressWarnings("JavadocMethod")
+  public NewKernelOperation(InputSocket.Factory inputSocketFactory,
+                            OutputSocket.Factory outputSocketFactory) {
+    this.typeSocket = inputSocketFactory.create(typeHint);
+    this.widthSocket = inputSocketFactory.create(widthHint);
+    this.heightSocket = inputSocketFactory.create(heightHint);
+    this.outputSocket = outputSocketFactory.create(outputHint);
+  }
+
+  @Override
+  public List<InputSocket> getInputSockets() {
+    return ImmutableList.of(
+            typeSocket,
+            widthSocket,
+            heightSocket
+    );
+  }
+
+  @Override
+  public List<OutputSocket> getOutputSockets() {
+    return ImmutableList.of(
+            outputSocket
+    );
+  }
+
+  @Override
+  public void perform() {
+    final int widthValue = widthSocket.getValue().get().intValue();
+    final int heightValue = heightSocket.getValue().get().intValue();
+    final int kernelType = typeSocket.getValue().get().value;
+
+    outputSocket.setValue(opencv_imgproc.getStructuringElement(kernelType, new Size(widthValue,
+                          heightValue)));
```

Comment:
Won't this allocate a new matrix every time this perform method is called.
Don't you instead want to pass in an existing matrix so the memory can be reused?

Reply:
You're right. Ideally you'd create the kernel once, but idk how to get that behavior..

Reply:
There should be an overload that takes a `dest` as a parameter.

Reply:
I'm not sure I understand.

Reply:
It's december of the next year, which means it's my turn to comment 😂 

Going back and looking through this, I understand what we are going for, but not sure how that functionality is created in GRIP. The kernel creating isn't part of the pipeline, just the setup, although I think i'm combining the code generation with the actual application..lemme see what I can come up with for the application part and then I'll worry about code gen later